src/splinterdb.c

```diff
@@ -785,3 +789,57 @@ splinterdb_close_print_stats(splinterdb *kvs)
    task_print_stats(kvs->task_sys);
    splinterdb_stats_print_insertion(kvs);
 }
+
+/*
+ * -------------------------------------------------------------------------
+ * External "APIs" provided mainly to invoke lower-level functions intended
+ * for use -ONLY- as testing interfaces.
+ * -------------------------------------------------------------------------
+ */
+void
+splinterdb_cache_flush(const splinterdb *kvs)
```

Comment:
@rtjohnso : Review why this externalization is needed.

And what is the intent behind the usage of this interface in the newly added `test_completion_of_outstanding_async_IOs_from_process_bug()` test case.

Reply:
Revisited both newly added test case:

- `test_completion_of_outstanding_async_IOs_from_process_bug()`
- `test_multiple_forked_process_doing_IOs()`

The point of using `cache_flush()` is so that we can invoke the call to `io_cleanup_all()` -> `io_cleanup()` to complete all pending AIOs.

Both test cases essentially are exercisers for these interfaces to cross-check the behaviour of whether we can issue any IO-cleanup related calls after thread deregistration.

The tests are somewhat of a negative test in that they verify an expected code-flow of execution.

These test cases seem useful as a way of self-documenting stabilization test cases, hence, I'd like to retain these.